crates/fuel-core/src/service/adapters.rs

```diff
@@ -318,23 +320,49 @@ impl TransactionsSource {
     }
 }
 
+pub struct NewTxWaiter {
+    pub receiver: tokio::sync::watch::Receiver<()>,
+    pub timeout: Instant,
+}
+
+impl NewTxWaiter {
+    pub fn new(receiver: tokio::sync::watch::Receiver<()>, timeout: Instant) -> Self {
+        Self { receiver, timeout }
+    }
+}
+
+#[derive(Clone)]
+pub struct PreconfirmationSender {
+    pub sender: tokio::sync::mpsc::Sender<Vec<PreconfirmationStatus>>,
+}
+
+impl PreconfirmationSender {
+    pub fn new(sender: tokio::sync::mpsc::Sender<Vec<PreconfirmationStatus>>) -> Self {
+        Self { sender }
+    }
+}
+
 #[derive(Clone)]
 pub struct ExecutorAdapter {
     pub(crate) executor: Arc<Executor<Database, Database<Relayer>>>,
+    pub new_txs_watcher: tokio::sync::watch::Receiver<()>,
+    pub preconfirmation_sender: PreconfirmationSender,
 }
 
 impl ExecutorAdapter {
     pub fn new(
         database: Database,
         relayer_database: Database<Relayer>,
-        // #[cfg(feature = "parallel-executor")]
-        // config: fuel_core_parallel_executor::config::Config,
-        // #[cfg(not(feature = "parallel-executor"))]
         config: fuel_core_upgradable_executor::config::Config,
+        new_txs_watcher: tokio::sync::watch::Receiver<()>,
+        preconfirmation_sender: tokio::sync::mpsc::Sender<Vec<PreconfirmationStatus>>,
```

Comment:
Should this be depending on the ports instead of these concrete types?

I'll keep reading to understand better this flow.

It feels weird to have a "port" for something we construct concretely (`PreconfirmationSender`), maybe it's not a port but just an abstraction?

Reply:
```rs
pub struct BlockExecutor<R, TxWaiter, PreconfirmationSender> {
    relayer: R,
    consensus_params: ConsensusParameters,
    options: ExecutionOptions,
    new_tx_waiter: TxWaiter,
    preconfirmation_sender: PreconfirmationSender,
}
```

I think we need better names for these things.

But it looks like maybe we are just integrating things here.

Reply:
It is an adapter, why do we need to have a port here?